### DISCUSSION TITLE
How to make tokenizer handle double-word countries name ?

### DISCUSSION THREAD

**Discussion by ToSev7en** [explosion/spaCy#2793] [Help: Coding & Implementations] [answered]
Such as "south korea is an country where..." , the tokenizer usually return two tokens: "south" and  "korea"

but what I want is "south korea", just one token is better for my problem.

Thanks your attention~

## Info about spaCy
* **spaCy version:** 2.0.11
* **Platform:** Darwin-17.7.0-x86_64-i386-64bit
* **Python version:** 3.6.2
* **Models:** en

**Comment by ines** [comment, marked as answer]
If you want "South Korea" to be one token, the best approach would be to find and then *merge* the tokens afterwards. You can do this by adding a custom component to your pipeline.

This example shows a pretty similar use case: https://github.com/explosion/spaCy/blob/master/examples/pipeline/custom_component_countries_api.py

Given a list of countries, it uses the `PhraseMatcher` to find them in the `Doc` and merges them into one token. Optionally, you can also set entity labels or custom attributes on the merged spans.
